lib/chewy/type/import/bulk.rb

```diff
@@ -0,0 +1,76 @@
+module Chewy
+  class Type
+    module Import
+      # Adds additional features to elasticsearch-api bulk method:
+      # * supports Chewy index suffix if necessary;
+      # * supports bulk_size, devides the passed body in chunks
+      #   and peforms a separate request for each chunk;
+      # * returns only errored document entries from the response
+      #   if any present.
+      #
+      # @see https://github.com/elastic/elasticsearch-ruby/blob/master/elasticsearch-api/lib/elasticsearch/api/actions/bulk.rb
+      class Bulk
+        # @param type [Chewy::Type] a type for the request
+        # @param suffix [String] an index name optional suffix
+        # @param bulk_size [Integer] bulk size in bytes
+        # @param bulk_options [Hash] options passed to the elasticsearch-api bulk method
+        def initialize(type, suffix: nil, bulk_size: nil, **bulk_options)
+          @type = type
+          @suffix = suffix
+          @bulk_size = bulk_size - 1.kilobyte if bulk_size # 1 kilobyte for request header and newlines
+          @bulk_options = bulk_options
+
+          raise ArgumentError, '`bulk_size` can\'t be less than 1 kilobyte' if @bulk_size && @bulk_size <= 0
+        end
+
+        # Performs a bulk request with the passed body, returns empty
+        # array if everything is fine and array filled with errored
+        # document entries if something went wrong.
+        #
+        # @param body [Array<Hash>] a standard bulk request body
+        # @return [Array<Hash>] an array of bulk errors
+        def perform(body)
```

Comment:
according to the doc `body` parameter is an array of hashes. so, it would be better to name it `bodies` or maybe even better `bulk_pieces` or `bulk_parts`. WDYT?